packages/block-editor/src/components/block-list/index.js

```diff
@@ -41,13 +46,55 @@ import {
 	DEFAULT_BLOCK_EDIT_CONTEXT,
 } from '../block-edit/context';
 
-const elementContext = createContext();
+const componentsContext = createContext();
 
 export const IntersectionObserver = createContext();
 const pendingBlockVisibilityUpdatesPerRegistry = new WeakMap();
 
+function useRootPortal( component ) {
+	const components = useContext( componentsContext );
+
+	useLayoutEffect( () => {
+		if ( ! components ) return;
+		components.add( component );
+		return () => {
+			components.delete( component );
+		};
+	} );
```

Comment:
This seems fragile. Every time it renders it deletes the component and adds a new one. And when it does that it changes the order of the components. Because of the lack of keys, it will just re-render everything.

Reply:
We can add keys 👍